packages/common/src/ui/layout/tables/components/DataRow/DataRow.tsx

```diff
@@ -1,68 +1,102 @@
-import React from 'react';
+import React, { FC, useCallback } from 'react';
 import TableRow from '@mui/material/TableRow';
 import TableCell from '@mui/material/TableCell';
 import { Column } from '../../columns/types';
 import { DomainObject } from '../../../../../types';
+import { TableStore, useTableStore } from '../..';
+import { Collapse } from '@mui/material';
 
 interface DataRowProps<T extends DomainObject> {
   columns: Column<T>[];
   onClick?: (rowValues: T) => void;
   rowData: T;
   rowKey: string;
+  ExpandContent?: FC;
 }
 
+const useExpanded = (rowId: string) => {
+  const selector = useCallback(
+    (state: TableStore) => {
+      return {
+        rowId,
+        isExpanded: state.rowState[rowId]?.isExpanded,
+        toggleExpanded: () => state.toggleExpanded(rowId),
+      };
+    },
+    [rowId]
+  );
+
+  const equalityFn = (
+    oldState: ReturnType<typeof selector>,
+    newState: ReturnType<typeof selector>
+  ) =>
+    oldState?.isExpanded === newState?.isExpanded &&
+    oldState.rowId === newState.rowId;
+
+  const { isExpanded, toggleExpanded } = useTableStore(selector, equalityFn);
+
+  return { isExpanded, toggleExpanded };
+};
```

Comment:
this is duplicated and doesn't need to be - think I will put it in the `TableContext` folder